db/query.go

```diff
@@ -238,11 +238,12 @@ func (context *DatabaseContext) N1QLQueryWithStats(queryName string, statement s
 
 	results, err = gocbBucket.Query(statement, params, consistency, adhoc)
 	if err != nil {
-		context.DbStats.StatsGsiViews().Add(fmt.Sprintf(base.StatKeyN1qlQueryErrorCountExpvarFormat, queryName), 1)
+		context.DbStats.NewStats.GSIStats(queryName).QueryErrorCount.Add(1)
 	}
 
-	context.DbStats.StatsGsiViews().Add(fmt.Sprintf(base.StatKeyN1qlQueryCountExpvarFormat, queryName), 1)
-	context.DbStats.StatsGsiViews().Add(fmt.Sprintf(base.StatKeyN1qlQueryTimeExpvarFormat, queryName), time.Since(startTime).Nanoseconds())
+	context.DbStats.NewStats.GSIStats(queryName).QueryErrorCount.Add(1)
+	context.DbStats.NewStats.GSIStats(queryName).QueryTime.Add(int(time.Since(startTime).Nanoseconds()))
```

Comment:
An advantage of this approach is that we can just call context.DbStats.NewStats.GSIStats(queryName) once in this method (to minimize map lookups), and use that in case there are multiple stat updates.